Zend/zend_generators.c

```diff
@@ -784,9 +785,12 @@ ZEND_API void zend_generator_resume(zend_generator *orig_generator) /* {{{ */
 		return;
 	}
 
-	if (UNEXPECTED((orig_generator->flags & ZEND_GENERATOR_DO_INIT) != 0 && !Z_ISUNDEF(generator->value))) {
-		/* We must not advance Generator if we yield from a Generator being currently run */
-		return;
+	if (UNEXPECTED(orig_generator->flags & ZEND_GENERATOR_DO_INIT)) {
+		orig_generator->flags &= ~ZEND_GENERATOR_DO_INIT;
+		if (!Z_ISUNDEF(generator->value)) {
```

Comment:
This check is bogus … because the direct yielded generator might not be primed yet (i.e. not have a value), but the inner generator (i.e. see the test) may.

Reply:
I don't get what you mean. The value is only stored on the inner generator. This check matches what was there before as well.